robot/impl/optional_dependencies_test.go

```diff
@@ -619,3 +619,218 @@ func TestModularOptionalDependencies(t *testing.T) {
 		test.That(t, doCommandResp, test.ShouldResemble, map[string]any{"optional_motor_state": "moving: false"})
 	}
 }
+
+// Contains _another_ MOC that this MOC will optionally depend upon.
+type mutualOptionalChildConfig struct {
+	OtherMOC string `json:"other_moc"`
+}
+
+// Validate validates the config and returns an optional dependency on `other_moc`.
+func (mocCfg *mutualOptionalChildConfig) Validate(path string) ([]string, []string, error) {
+	if mocCfg.OtherMOC == "" {
+		return nil, nil,
+			fmt.Errorf(`expected "other_moc" attribute for MOC %q`, path)
+	}
+	return nil, []string{mocCfg.OtherMOC}, nil
+}
+
+type mutualOptionalChild struct {
+	resource.Named
+	resource.TriviallyCloseable
+	logger logging.Logger
+
+	otherMOC      resource.Resource
+	reconfigCount int
+}
+
+func newMutualOptionalChild(ctx context.Context,
+	deps resource.Dependencies,
+	conf resource.Config,
+	logger logging.Logger,
+) (*mutualOptionalChild, error) {
+	moc := &mutualOptionalChild{
+		Named:  conf.ResourceName().AsNamed(),
+		logger: logger,
+	}
+
+	if err := moc.Reconfigure(ctx, deps, conf); err != nil {
+		return nil, err
+	}
+
+	return moc, nil
+}
+
+func (moc *mutualOptionalChild) Reconfigure(ctx context.Context, deps resource.Dependencies,
```

Comment:
Discussed offline: you _must_ have a reconfigurable resource to leverage optional dependencies at all. This is not something that was captured in documentation, so I'll have to chat with NetCode/docs about this.